### DISCUSSION TITLE
Is it possbile to change or access decorator variables?

### DISCUSSION THREAD

**Discussion by fendyk** [instantcommerce/next-api-decorators#539] [Q&A] [answered]
Let's say I have a list function like this:

```
@Get()
async list() {
    await db.users.findMany();
}
```

And now, I need to add a header with 'X-total-Count'. How would I approach this using a decorator? For example, I currently use the following:

```
@Get()
async list(@Res() res: NextApiResponse) {
    const items = await db.user.findMany();
    setTotalCountHeader(res, items.length) // translates to: res.setHeader('X-Total-Count', amount)
    return items;
}
```

This is a workaround and it works as aspected, but could I use it with a decorator? The only thing I can think of is the following:
```
@Get()
@SetHeader('X-Total-Count', '2') // Value needs to be pulled from somewhere in the function below
async list(@Res() res: NextApiResponse) {
    const items = await db.user.findMany();
    return items;
}
```

Is this just not possbile with decorators? Let me know, thanks!

**Comment by ggurkal** [comment, marked as answer]
Hi @fendyk 

Unfortunately, this is not possible.
